src/tensors/tensor.jl

```diff
@@ -495,105 +479,32 @@ See also [`Base.setindex!(::TensorMap{T,S,N₁,N₂}, ::Any, ::FusionTree{I,N₁
 end
 
-"""
-    Base.setindex!(t::TensorMap{T,S,N₁,N₂,I},
-                   v,
-                   f₁::FusionTree{I,N₁},
-                   f₂::FusionTree{I,N₂}) where {T,S,N₁,N₂,I<:Sector}
-    t[f₁, f₂] = v
-
-Copies `v` into the  data slice of `t` corresponding to the splitting - fusion tree pair
-`(f₁, f₂)`. Here, `v` can be any object that can be copied into a `StridedViews.StridedView`
-of size `(dims(codomain(t), f₁.uncoupled)..., dims(domain(t), f₂.uncoupled))` using
-`Base.copy!`.
-
-See also [`Base.getindex(::TensorMap{T,S,N₁,N₂}, ::FusionTree{I,N₁}, ::FusionTree{I,N₂}) where {T,S,N₁,N₂,I<:Sector}`](@ref)
-"""
-@propagate_inbounds function Base.setindex!(
-        t::TensorMap{T, S, N₁, N₂}, v, f₁::FusionTree{I, N₁}, f₂::FusionTree{I, N₂}
-    ) where {T, S, N₁, N₂, I <: Sector}
-    return copy!(getindex(t, f₁, f₂), v)
-end
-
-"""
-    Base.getindex(t::TensorMap
-                  sectors::NTuple{N₁+N₂,I}) where {N₁,N₂,I<:Sector} 
-        -> StridedViews.StridedView
-    t[sectors]
-
-Return a view into the data slice of `t` corresponding to the splitting - fusion tree pair
-with combined uncoupled charges `sectors`. In particular, if `sectors == (s₁..., s₂...)`
-where `s₁` and `s₂` correspond to the uncoupled charges in the codomain and domain
-respectively, then a `StridedViews.StridedView` of size
-`(dims(codomain(t), s₁)..., dims(domain(t), s₂))` is returned.
-
-This method is only available for the case where `FusionStyle(I) isa UniqueFusion`,
-since it assumes a  uniquely defined coupled charge.
-"""
-@inline function Base.getindex(t::TensorMap, sectors::Tuple{I, Vararg{I}}) where {I <: Sector}
-    I === sectortype(t) || throw(SectorMismatch("Not a valid sectortype for this tensor."))
-    FusionStyle(I) isa UniqueFusion ||
-        throw(SectorMismatch("Indexing with sectors only possible if unique fusion"))
-    length(sectors) == numind(t) ||
-        throw(ArgumentError("Number of sectors does not match."))
-    s₁ = TupleTools.getindices(sectors, codomainind(t))
-    s₂ = map(dual, TupleTools.getindices(sectors, domainind(t)))
-    c1 = length(s₁) == 0 ? unit(I) : (length(s₁) == 1 ? s₁[1] : first(⊗(s₁...)))
-    @boundscheck begin
-        c2 = length(s₂) == 0 ? unit(I) : (length(s₂) == 1 ? s₂[1] : first(⊗(s₂...)))
-        c2 == c1 || throw(SectorMismatch("Not a valid sector for this tensor"))
-        hassector(codomain(t), s₁) && hassector(domain(t), s₂)
-    end
-    f₁ = FusionTree(s₁, c1, map(isdual, tuple(codomain(t)...)))
-    f₂ = FusionTree(s₂, c1, map(isdual, tuple(domain(t)...)))
-    @inbounds begin
-        return t[f₁, f₂]
-    end
-end
-@propagate_inbounds function Base.getindex(t::TensorMap, sectors::Tuple)
-    return t[map(sectortype(t), sectors)]
-end
-
 # Show
 #------
-function Base.summary(io::IO, t::TensorMap)
-    return print(io, "TensorMap(", space(t), ")")
+function type_repr(::Type{TensorMap{T, S, N₁, N₂, A}}) where {T, S, N₁, N₂, A}
```

Comment:
`type_repr` comes from TensorKitSectors.jl, right? Do we import this somewhere? Or does TensorKit.jl has its own `type_repr` (which doesn't sound right).

Reply:
We do: 

https://github.com/QuantumKitHub/TensorKit.jl/blob/9ba3b6c7ad4738ad9b25d6f2527a9a2767927a4a/src/TensorKit.jl#L122

Happy to qualify if you think that is more obvious, but I think it is really fair to say we "take ownership" of these functions in TensorKitSectors so this is kind of fine to me

Reply:
Ok thanks; I had checked before posing the question, but had overlooked (internet is working reasonably well but not superfast on the train). I am fine with keeping it like this.